force-app/main/default/classes/SOQL.cls

```diff
@@ -314,28 +460,61 @@ public with sharing class SOQL {
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
-
+        List<OpportunityLineItem> originalLineItems = [SELECT Product2Id, Quantity, UnitPrice, PricebookEntryId
+                                                        FROM OpportunityLineItem 
+                                                        WHERE OpportunityId = :origOppId];
+    
+        // Create a map to hold PricebookEntryId to Pricebook2Id
+        Map<Id, Id> pricebookEntryToPricebookMap = new Map<Id, Id>();
+        Set<Id> pricebookEntryIds = new Set<Id>();
+    
+        // Collect all PricebookEntryIds from original line items
+        for (OpportunityLineItem oli : originalLineItems) {
+            pricebookEntryIds.add(oli.PricebookEntryId);
+        }
+    
+        // Query PricebookEntry records to get the Pricebook2Id
+        for (PricebookEntry pbe : [SELECT Id, Pricebook2Id FROM PricebookEntry WHERE Id IN :pricebookEntryIds]) {
+            pricebookEntryToPricebookMap.put(pbe.Id, pbe.Pricebook2Id);
+        }
+    
         // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
-        // Loop through each cloned Opportunity Id
-
+        List<OpportunityLineItem> newLineItems = new List<OpportunityLineItem>();
+    
+        for (Id cloneOppId : cloneOppIds) {
+            // Ensure the Opportunity has a Pricebook assigned
+            // (Assuming that the cloned Opportunities have Pricebooks assigned as in the test class)
+            // This is not handled here as we assume the test data has this already set.
+    
             // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
-
-        // Insert the new line items
-    }
+            for (OpportunityLineItem origLineItem : originalLineItems) {
+                OpportunityLineItem newLineItem = new OpportunityLineItem();
+                newLineItem.OpportunityId = cloneOppId;
+                newLineItem.Product2Id = origLineItem.Product2Id;
+                newLineItem.Quantity = origLineItem.Quantity + Integer.valueOf(Math.floor(Math.random() * 3) + 1); // Add the random number to the quantity
```

Comment:
Technically, you don't need the `Math.floor()` method here since casting an Integer rounds off the number anyway.